src/main/java/org/folio/inventory/support/KafkaConsumerVerticle.java

```diff
@@ -0,0 +1,144 @@
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_MAX_REQUEST_SIZE;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_PORT;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_REPLICATION_FACTOR;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.OKAPI_URL;
+import static org.folio.kafka.KafkaTopicNameHelper.createSubscriptionDefinition;
+import static org.folio.kafka.KafkaTopicNameHelper.getDefaultNameSpace;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Verticle;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonObject;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.logging.log4j.Logger;
+import org.folio.inventory.dataimport.cache.MappingMetadataCache;
+import org.folio.inventory.storage.Storage;
+import org.folio.kafka.GlobalLoadSensor;
+import org.folio.kafka.KafkaConfig;
+import org.folio.kafka.KafkaConsumerWrapper;
+import org.folio.kafka.SubscriptionDefinition;
+import org.folio.okapi.common.GenericCompositeFuture;
+
+public abstract class KafkaConsumerVerticle extends AbstractVerticle {
+  private static final String LOAD_LIMIT_TEMPLATE = "inventory.kafka.%s.loadLimit";
+  private static final String LOAD_LIMIT_DEFAULT = "5";
+  private static final String MAX_DISTRIBUTION_NUMBER_TEMPLATE = "inventory.kafka.%s.maxDistributionNumber";
+  private static final String MAX_DISTRIBUTION_NUMBER_DEFAULT = "100";
+  private static final String CACHE_EXPIRATION_DEFAULT = "3600";
+  private static final String METADATA_EXPIRATION_TIME = "inventory.mapping-metadata-cache.expiration.time.seconds";
+  private final List<KafkaConsumerWrapper<String, String>> consumerWrappers = new ArrayList<>();
+  private KafkaConfig kafkaConfig;
+  private JsonObject config;
+  private HttpClient httpClient;
+  private Storage storage;
+  private MappingMetadataCache mappingMetadataCache;
+
+  @Override
+  public void stop(Promise<Void> stopPromise) {
+    List<Future<Void>> stopFutures = consumerWrappers.stream()
+      .map(KafkaConsumerWrapper::stop)
+      .toList();
+
+    GenericCompositeFuture.join(stopFutures)
+      .onComplete(ar -> stopPromise.complete());
+  }
+
+  protected abstract Logger getLogger();
+
+  protected KafkaConsumerWrapper<String, String> createConsumer(String eventType) {
+    var kafkaConsumerWrapper = KafkaConsumerWrapper.<String, String>builder()
+      .context(context)
+      .vertx(vertx)
+      .kafkaConfig(getKafkaConfig())
+      .loadLimit(getLoadLimit())
+      .globalLoadSensor(new GlobalLoadSensor())
+      .subscriptionDefinition(getSubscriptionDefinition(getKafkaConfig().getEnvId(), eventType))
+      .build();
+    consumerWrappers.add(kafkaConsumerWrapper);
+    return kafkaConsumerWrapper;
+  }
+
+  protected JsonObject getConfig() {
+    return ofNullable(config).orElseGet(() -> {
```

Comment:
The parent class `AbstractVerticle` already has a `config()` method. Should we use that instead of creating this new method?

Reply:
I think we should not use super method, because all 4 existed ConsumerVerticles used to have it as:
`JsonObject config = vertx.getOrCreateContext().config();`
and we basically do the same or return previously got config